sdk/include/opentelemetry/sdk/metrics/state/multi_metric_storage.h

```diff
@@ -18,46 +19,57 @@ namespace metrics
 class MultiMetricStorage : public WritableMetricStorage
 {
 public:
+  MultiMetricStorage(nostd::shared_ptr<ExemplarReservoir> exemplar_reservoir)
+      : exemplar_reservoir_(exemplar_reservoir)
+  {}
+
   void AddStorage(std::shared_ptr<WritableMetricStorage> storage) { storages_.push_back(storage); }
 
-  virtual void RecordLong(long value) noexcept override
+  virtual void RecordLong(long value,
+                          const opentelemetry::context::Context &context) noexcept override
   {
+    exemplar_reservoir_->OfferMeasurement(value, {}, context);
     for (auto &s : storages_)
     {
-      s->RecordLong(value);
+      s->RecordLong(value, context);
     }
   }
 
-  virtual void RecordLong(
-      long value,
-      const opentelemetry::common::KeyValueIterable &attributes) noexcept override
+  virtual void RecordLong(long value,
+                          const opentelemetry::common::KeyValueIterable &attributes,
+                          const opentelemetry::context::Context &context) noexcept override
   {
+    exemplar_reservoir_->OfferMeasurement(value, attributes, context);
```

Comment:
Should this method be called here, or just inside all the storage?

Reply:
removed from the `MultiMetricStorage`.